canvas/animation.go

```diff
@@ -22,17 +22,20 @@ func NewColorRGBAAnimation(start, stop color.Color, d time.Duration, fn func(col
 	r1, g1, b1, a1 := start.RGBA()
 	r2, g2, b2, a2 := stop.RGBA()
 
-	rDelta := int(r2) - int(r1)
-	gDelta := int(g2) - int(g1)
-	bDelta := int(b2) - int(b1)
-	aDelta := int(a2) - int(a1)
+	rStart := uint8(r1 >> 8)
+	gStart := uint8(g1 >> 8)
+	bStart := uint8(b1 >> 8)
+	aStart := uint8(a1 >> 8)
+	rDelta := float32(int(r2>>8) - int(rStart))
+	gDelta := float32(int(g2>>8) - int(gStart))
+	bDelta := float32(int(b2>>8) - int(bStart))
+	aDelta := float32(int(a2>>8) - int(aStart))
```

Comment:
The start values are always converted to `int` before being used. Therefore, they should be stored as `int` already.

And regarding the delta values for animation: Storing them as `float32` saves conversion during animation and does not use up more space than `int`.